### ISSUE TITLE
how to configure to run in a subdir?

### ISSUE THREAD

**Issue by junalmeida** [tiredofit/docker-freepbx#9]
is there a way to set this image to run under a subdir, such as http://example.com/pbx

**Comment by tiredofit** [comment]
I just comitted a new release `1.1` which fixes a few bugs that I have encountered in the past few weeks trying to get UCP working, I also added the new environment variable `WEBROOT`. You would use it like so: `WEBROOT=/var/www/html/pbx` which should give you what you are after! Do let me know if it is working for you. The Docker Hub Image will take about 6 hours to appear.

**Comment by junalmeida** [comment]
Thank you for your rapid response. I am going to try this out soon!

**Comment by junalmeida** [comment]
I did a docker-compose down; docker-compose pull, and it downloaded a newer image... then I deleted previous directories, and edited my docker-compose like the following:
```
freepbx-app:
  container_name: freepbx-app
  hostname: mydomain
  image: 'tiredofit/freepbx:latest'
  ports:
    - 5060:5060
    - 5160:5160
    - 18000:20000/udp
    - 805:80
  volumes:
    - ./certs:/certs
    - ./data:/data
    - ./logs:/var/log/asterisk
    - ./data/www:/var/www/html
    - ./db:/var/lib/mysql
    - './configure.sh:/usr/local/bin/configure.sh'
  environment:
    - VIRTUAL_HOST=mydomain
    - VIRTUAL_PORT=80
    - ZABBIX_HOSTNAME=freepbx-app
    - RTP_START=18000
    - RTP_FINISH=20000
    - DB_EMBEDDED=TRUE
    - WEBROOT=/var/www/html/pbx
  restart: always
  cap_add:
    - NET_ADMIN

```

But reaching from the browser I got:

```
Not Found
The requested URL /pbx/ was not found on this server.

Apache/2.4.25 (Debian) Server at
```

**Comment by junalmeida** [comment]
I checked /var/www/html and all files are in the root, there is no pbx dir inside html.

**Comment by tiredofit** [comment]
Nuts. Back to drawing board for me. Will check in in ~ 2 hours with thoughts.

**Comment by tiredofit** [comment]
@junalmeida I just tried a fresh install with the following set in my docker-compose.yml

`      - WEBROOT=/var/www/html/pbx`  

And it seems to be working without problems. If you have an existing install, perhaps take a copy of `/data/www` for backup, and then just create a `/data/www/pbx` moving all files inside of it?

**Comment by junalmeida** [comment]
Maybe I have a tip - amportal.conf: 
```amportal.conf
# FreePBX Web Root Dir
# Default Value: /var/www/html
AMPWEBROOT=/var/www/html
```

**Comment by junalmeida** [comment]
@tiredofit I did a docker-compose down, and deleted all directories, fresh install here I suppose :(

**Comment by junalmeida** [comment]
What I did so far: 
```
root@alma-dev:/data/freepbx# ls
certs  data  db  docker-compose.yml  logs
root@alma-dev:/data/freepbx# docker-compose down
Stopping freepbx-app ... done
Removing freepbx-app ... done
root@alma-dev:/data/freepbx# rm -rf certs data db logs
root@alma-dev:/data/freepbx# ls
docker-compose.yml
root@alma-dev:/data/freepbx# docker-compose up -d
Creating freepbx-app
```

**Comment by junalmeida** [comment]
Then analysing the logs:
 ```
Downloading module 'framework'
freepbx-app    | Processing framework
freepbx-app    | Downloading...
freepbx-app    |         0/11703492 [>---------------------------]   0%
freepbx-app    |   1176464/11703492 [==>-------------------------]  10%
freepbx-app    |   2347472/11703492 [=====>----------------------]  20%
freepbx-app    |   3515728/11703492 [========>-------------------]  30%
freepbx-app    |   4682664/11703492 [===========>----------------]  40%
freepbx-app    |   5857664/11703492 [==============>-------------]  50%
freepbx-app    |   7027416/11703492 [================>-----------]  60%
freepbx-app    |   8192952/11703492 [===================>--------]  70%
freepbx-app    |   9366696/11703492 [======================>-----]  80%
freepbx-app    |  10537704/11703492 [=========================>--]  90%
freepbx-app    |
freepbx-app    | Finished downloading
freepbx-app    | Extracting...Done
freepbx-app    | Download completed in 14 seconds
freepbx-app    | Updating tables admin, ampusers, cronmanager, featurecodes, freepbx_log, freepbx_settings, globals, module_xml, modules, notifications...Done
freepbx-app    | installing files to /var/www/html..done
freepbx-app    | installing files to /var/lib/asterisk/bin..done
freepbx-app    | installing files to /var/lib/asterisk/agi-bin..done
freepbx-app    | Checking for upgrades..
freepbx-app    | No further upgrades necessary
freepbx-app    | framework file install done, removing packages from module
freepbx-app    | file/directory: /var/www/html/admin/modules/framework/amp_conf removed successfully
freepbx-app    | file/directory: /var/www/html/admin/modules/framework/upgrades removed successfully
freepbx-app    | file/directory: /var/www/html/admin/modules/framework/start_asterisk removed successfully
freepbx-app    | file/directory: /var/www/html/admin/modules/framework/install removed successfully
freepbx-app    | file/directory: /var/www/html/admin/modules/framework/installlib removed successfully
```

See the message "installing files to"

So strange that it works for you but not for me :(

**Comment by tiredofit** [comment]
Yes very strange. During my install I get the following:
 Downloading module 'framework'
````
bell-tiredofit-ca-app    | Processing framework
bell-tiredofit-ca-app    | Downloading...
bell-tiredofit-ca-app    |         0/11703492 [>---------------------------]   0%
bell-tiredofit-ca-db     | ** [cron] Starting cron
bell-tiredofit-ca-app    |   1173728/11703492 [==>-------------------------]  10%
bell-tiredofit-ca-app    |   2342016/11703492 [=====>----------------------]  20%
bell-tiredofit-ca-app    |   3514392/11703492 [========>-------------------]  30%
bell-tiredofit-ca-app    |   4682552/11703492 [===========>----------------]  40%
bell-tiredofit-ca-app    |   5853672/11703492 [==============>-------------]  50%
bell-tiredofit-ca-app    |   7028544/11703492 [================>-----------]  60%
bell-tiredofit-ca-app    |   8192952/11703492 [===================>--------]  70%
bell-tiredofit-ca-app    |   9369416/11703492 [======================>-----]  80%
bell-tiredofit-ca-app    | 
bell-tiredofit-ca-app    | Finished downloading
bell-tiredofit-ca-app    |  10533440/11703492 [=========================>--]  90%
bell-tiredofit-ca-app    | Extracting...Done
bell-tiredofit-ca-app    | Download completed in 16 seconds
bell-tiredofit-ca-app    | Updating tables admin, ampusers, cronmanager, featurecodes, freepbx_log, freepbx_settings, globals, module_xml, modules, notifications...Done
bell-tiredofit-ca-app    | installing files to /var/www/html/pbx..done
bell-tiredofit-ca-app    | installing files to /var/lib/asterisk/bin..done
bell-tiredofit-ca-app    | installing files to /var/lib/asterisk/agi-bin..done
````
Can I see a copy of your docker-compose with necessary juicy bits blanked out?

**Comment by junalmeida** [comment]
Sure: 

```
root@alma-dev:/data/freepbx# cat docker-compose.yml
freepbx-app:
  container_name: freepbx-app
  hostname: sis.ferloassessoria.com.br
  image: 'tiredofit/freepbx:latest'
  ports:
    - 5060:5060
    - 5160:5160
    - 18000:20000/udp
    - 805:80
  volumes:
    - ./certs:/certs
    - ./data:/data
    - ./logs:/var/log/asterisk
    #- ./data/www:/var/www/html
    - ./db:/var/lib/mysql
  environment:
    - WEBROOT=/var/www/html/pbx
    - VIRTUAL_HOST=sis.ferloassessoria.com.br
    - VIRTUAL_PORT=80
    #- LETSENCRYPT_HOST=sis.ferloassessoria.com.br
    #- LETSENCRYPT_EMAIL=email@example.com
    - ZABBIX_HOSTNAME=freepbx-app
    - RTP_START=18000
    - RTP_FINISH=20000
    - DB_EMBEDDED=TRUE
  restart: always
  cap_add:
    - NET_ADMIN

```

**Comment by tiredofit** [comment]
All looks good to me! Let's try a docker pull tiredofit/freepbx just in case?

**Comment by junalmeida** [comment]
```
root@alma-dev:/data/freepbx# docker pull tiredofit/freepbx
Using default tag: latest
latest: Pulling from tiredofit/freepbx
Digest: sha256:961ac1c60acaae250004792e3157a49bbdbc79fafc9137e655f74b9bbcfd5442
Status: Image is up to date for tiredofit/freepbx:latest
```

**Comment by tiredofit** [comment]
I found a small error in one of my scripts which might be impacting it. Trying to resubmit to Docker Hub however it is failing. If that is the case I will setup a temporary repo for you to test this change.

**Comment by junalmeida** [comment]
Ok, I cant test it!

**Comment by junalmeida** [comment]
I guess that I found the problem: 
```
if [ "$DB_EMBEDDED" = "FALSE" ];  then 
	./install -n --dbuser=$DB_USER --dbpass=$DB_PASS --dbname=$DB_NAME --cdrdbname=$DB_NAME --webroot=$WEBROOT
else
  ./install -n
fi
```

Since I am using the db embeeded, I was punished! lol

**Comment by tiredofit** [comment]
Good catch!. Let me add that.

**Comment by junalmeida** [comment]
Ok, i see your commit. Do you have a testing repo? Docker hub shows no updates.

**Comment by tiredofit** [comment]
Looks like the docker hub is blocking me from building as I'm on a free account and used too many of their resources. My other repo is locked behind a corporate LDAP server, so I am sending one up manually..

**Comment by tiredofit** [comment]
OK! there should be a new tiredofit/freepbx:latest or tiredofit/freepbx:1.2 available.

**Comment by junalmeida** [comment]
It works! 

Guess that you can close this issue. Thank you very much.

**Comment by tiredofit** [comment]
Thanks for your sleuthing to find where I went wrong!

**Comment by junalmeida** [comment]
@tiredofit Hello again. 

I did a simple change on the docker-compose file, just adjusting wrong ports. Then I called docker-compose down and up. This time I got an error, and 404 from the browser. Do you have any idea? 

```
Attaching to freepbx-app
freepbx-app    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
freepbx-app    | [s6-init] ensuring user provided files have correct perms...exited 0.
freepbx-app    | [fix-attrs.d] applying ownership & permissions fixes...
freepbx-app    | [fix-attrs.d] 02-zabbix: applying...
freepbx-app    | [fix-attrs.d] 02-zabbix: exited 0.
freepbx-app    | [fix-attrs.d] 03-logrotate: applying...
freepbx-app    | [fix-attrs.d] 03-logrotate: exited 0.
freepbx-app    | [fix-attrs.d] done.
freepbx-app    | [cont-init.d] executing container initialization scripts...
freepbx-app    | [cont-init.d] 01-permissions: executing...
freepbx-app    | **** [permissions] total 0 UIDs modified...
freepbx-app    | **** [permissions] total 0 GIDs modified...
freepbx-app    | [cont-init.d] 01-permissions: exited 0.
freepbx-app    | [cont-init.d] 02-zabbix: executing...
freepbx-app    | [cont-init.d] 02-zabbix: exited 0.
freepbx-app    | [cont-init.d] 03-cron: executing...
freepbx-app    | **** [cron] Disabling Cron
freepbx-app    | [cont-init.d] 03-cron: exited 0.
freepbx-app    | [cont-init.d] 04-fail2ban: executing...
freepbx-app    |
freepbx-app    | ** [fail2ban] Starting Fail2ban
freepbx-app    | Starting authentication failure monitor: fail2ban.
freepbx-app    | [cont-init.d] 04-fail2ban: exited 0.
freepbx-app    | [cont-init.d] 04-smtp: executing...
freepbx-app    | **** [smtp] Disabling SMTP Features
freepbx-app    | [cont-init.d] 04-smtp: exited 0.
freepbx-app    | [cont-init.d] 05-fail2ban: executing...
freepbx-app    |
freepbx-app    | ** [fail2ban] Starting Fail2ban
freepbx-app    | Starting authentication failure monitor: fail2ban.
freepbx-app    | [cont-init.d] 05-fail2ban: exited 0.
freepbx-app    | [cont-init.d] 09-mariadb: executing...
freepbx-app    | Stopping MariaDB database server: mysqld.
freepbx-app    | ** [mysql] New Embedded Database Detected, setting up..
freepbx-app    | Starting MariaDB database server: mysqld.
freepbx-app    | [cont-init.d] 09-mariadb: exited 0.
freepbx-app    | [cont-init.d] 10-freepbx: executing...
freepbx-app    | ** [freepbx] Setting File Permissions
freepbx-app    | ** [freepbx] New Install Detected - Please wait while we fetch FreePBX...
freepbx-app    |
freepbx-app    |
freepbx-app    | STARTING ASTERISK
freepbx-app    | Asterisk Started
freepbx-app    | ** [freepbx] Installing FreePBX
freepbx-app    | Assuming you are Database Root
freepbx-app    | Checking if SELinux is enabled...Its not (good)!
freepbx-app    | Reading /etc/asterisk/asterisk.conf...Done
freepbx-app    | Checking if Asterisk is running and we can talk to it as the 'asterisk' user...Yes. Determined Asterisk version to be: 14.7.6
freepbx-app    | Preliminary checks done. Starting FreePBX Installation
freepbx-app    | Checking if this is a new install...Yes (No /etc/freepbx.conf file detected)
freepbx-app    | Database Root installation checking credentials and permissions..Connected!
freepbx-app    | Initializing FreePBX Settings
freepbx-app    |
freepbx-app    | In utility.functions.php line 207:
freepbx-app    |
freepbx-app    |   fatal error reading freepbx_settings::
freepbx-app    |
freepbx-app    |
freepbx-app    | install [--dbengine DBENGINE] [--dbname DBNAME] [--cdrdbname CDRDBNAME] [--dbuser DBUSER] [--dbpass DBPASS] [--user USER] [--group GROUP] [--dev-links] [--skip-install] [--webroot WEBROOT] [--astetcdir ASTETCDIR] [--astmoddir ASTMODDIR] [--astvarlibdir ASTVARLIBDIR] [--astagidir ASTAGIDIR] [--astspooldir ASTSPOOLDIR] [--astrundir ASTRUNDIR] [--astlogdir ASTLOGDIR] [--ampbin AMPBIN] [--ampsbin AMPSBIN] [--ampcgibin AMPCGIBIN] [--ampplayback AMPPLAYBACK] [-r|--rootdb] [-f|--force]
freepbx-app    |
freepbx-app    | ** [freepbx] Enabling Default Modules
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 133: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 134: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 146: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 149: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 150: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 151: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 152: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 153: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 154: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 155: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 156: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 157: fwconsole: command not found
freepbx-app    | ** [freepbx] Starting Asterisk
freepbx-app    | [cont-init.d] 10-freepbx: exited 0.
freepbx-app    | [cont-init.d] 99-container-init: executing...
freepbx-app    | [cont-init.d] 99-container-init: exited 0.
freepbx-app    | [cont-init.d] done.
freepbx-app    | [services.d] starting services
freepbx-app    | [services.d] done.
freepbx-app    |
freepbx-app    | **** [zabbix] Starting Zabbix Agent

```

**Comment by tiredofit** [comment]
Time to reopen this :)
Let me try some tests on my end with starts and stops and see if I can trap it.

**Comment by tiredofit** [comment]
I can't seem to recreate it on my end, but did find that when visiting the root url without the subfolder indexing was enabled. I've added something to disable that functionality with latest image. But back to your problem:
It looks as if somehow it reset and tried to do an install one more time again and failed miserably.
`DEBUG=TRUE` might be a good setting to set to trace through each step of the script to see what went wrong perhaps?

**Comment by junalmeida** [comment]
Going to set debug env in 30 min and then I get back to you

**Comment by junalmeida** [comment]
Guess that I got nothing different using DEBUG=TRUE

```
root@alma-dev:/data/freepbx# docker-compose down
Stopping freepbx-app ... done
Removing freepbx-app ... done
root@alma-dev:/data/freepbx# nano docker-compose.yml
root@alma-dev:/data/freepbx# docker-compose up -d && docker-compose logs -f
Creating freepbx-app
Attaching to freepbx-app
freepbx-app    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
freepbx-app    | [s6-init] ensuring user provided files have correct perms...exited 0.
freepbx-app    | [fix-attrs.d] applying ownership & permissions fixes...
freepbx-app    | [fix-attrs.d] 02-zabbix: applying...
freepbx-app    | [fix-attrs.d] 02-zabbix: exited 0.
freepbx-app    | [fix-attrs.d] 03-logrotate: applying...
freepbx-app    | [fix-attrs.d] 03-logrotate: exited 0.
freepbx-app    | [fix-attrs.d] done.
freepbx-app    | [cont-init.d] executing container initialization scripts...
freepbx-app    | [cont-init.d] 01-permissions: executing...
freepbx-app    | **** [permissions] total 0 UIDs modified...
freepbx-app    | **** [permissions] total 0 GIDs modified...
freepbx-app    | [cont-init.d] 01-permissions: exited 0.
freepbx-app    | [cont-init.d] 02-zabbix: executing...
freepbx-app    | [cont-init.d] 02-zabbix: exited 0.
freepbx-app    | [cont-init.d] 03-cron: executing...
freepbx-app    | **** [cron] Disabling Cron
freepbx-app    | [cont-init.d] 03-cron: exited 0.
freepbx-app    | [cont-init.d] 04-fail2ban: executing...
freepbx-app    |
freepbx-app    | ** [fail2ban] Starting Fail2ban
freepbx-app    | Starting authentication failure monitor: fail2ban.
freepbx-app    | [cont-init.d] 04-fail2ban: exited 0.
freepbx-app    | [cont-init.d] 04-smtp: executing...
freepbx-app    | **** [smtp] Disabling SMTP Features
freepbx-app    | [cont-init.d] 04-smtp: exited 0.
freepbx-app    | [cont-init.d] 05-fail2ban: executing...
freepbx-app    |
freepbx-app    | ** [fail2ban] Starting Fail2ban
freepbx-app    | Starting authentication failure monitor: fail2ban.
freepbx-app    | [cont-init.d] 05-fail2ban: exited 0.
freepbx-app    | [cont-init.d] 09-mariadb: executing...
freepbx-app    | Stopping MariaDB database server: mysqld.
freepbx-app    | ** [mysql] New Embedded Database Detected, setting up..
freepbx-app    | Starting MariaDB database server: mysqld.
freepbx-app    | [cont-init.d] 09-mariadb: exited 0.
freepbx-app    | [cont-init.d] 10-freepbx: executing...
freepbx-app    | ** [freepbx] Setting File Permissions
freepbx-app    | ** [freepbx] New Install Detected - Please wait while we fetch FreePBX...
freepbx-app    |
freepbx-app    |
freepbx-app    | STARTING ASTERISK
freepbx-app    | Asterisk Started
freepbx-app    | ** [freepbx] Installing FreePBX
freepbx-app    | Assuming you are Database Root
freepbx-app    | Checking if SELinux is enabled...Its not (good)!
freepbx-app    | Reading /etc/asterisk/asterisk.conf...Done
freepbx-app    | Checking if Asterisk is running and we can talk to it as the 'asterisk' user...Yes. Determined Asterisk version to be: 14.7.6
freepbx-app    | Preliminary checks done. Starting FreePBX Installation
freepbx-app    | Checking if this is a new install...Yes (No /etc/freepbx.conf file detected)
freepbx-app    | Database Root installation checking credentials and permissions..Connected!
freepbx-app    | Initializing FreePBX Settings
freepbx-app    |
freepbx-app    | In utility.functions.php line 207:
freepbx-app    |
freepbx-app    |   fatal error reading freepbx_settings::
freepbx-app    |
freepbx-app    |
freepbx-app    | install [--dbengine DBENGINE] [--dbname DBNAME] [--cdrdbname CDRDBNAME] [--dbuser DBUSER] [--dbpass DBPASS] [--user USER] [--group GROUP] [--dev-links] [--skip-install] [--webroot WEBROOT] [--astetcdir ASTETCDIR] [--astmoddir ASTMODDIR] [--astvarlibdir ASTVARLIBDIR] [--astagidir ASTAGIDIR] [--astspooldir ASTSPOOLDIR] [--astrundir ASTRUNDIR] [--astlogdir ASTLOGDIR] [--ampbin AMPBIN] [--ampsbin AMPSBIN] [--ampcgibin AMPCGIBIN] [--ampplayback AMPPLAYBACK] [-r|--rootdb] [-f|--force]
freepbx-app    |
freepbx-app    | ** [freepbx] Enabling Default Modules
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 133: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 134: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 146: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 149: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 150: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 151: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 152: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 153: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 154: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 155: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 156: fwconsole: command not found
freepbx-app    | /var/run/s6/etc/cont-init.d/10-freepbx: line 157: fwconsole: command not found
freepbx-app    | ** [freepbx] Starting Asterisk
freepbx-app    | [cont-init.d] 10-freepbx: exited 0.
freepbx-app    | [cont-init.d] 99-container-init: executing...
freepbx-app    | [cont-init.d] 99-container-init: exited 0.
freepbx-app    | [cont-init.d] done.
freepbx-app    | [services.d] starting services
freepbx-app    | [services.d] done.
freepbx-app    |
freepbx-app    | **** [zabbix] Starting Zabbix Agent

```

**Comment by junalmeida** [comment]
I hope that the mysql db is intact, because I 've done all the extensions and trunks setup.

**Comment by tiredofit** [comment]
Hmm DEBUG would have been much more verbose.
This is what is concerning me:
`** [freepbx] New Install Detected - Please wait while we fetch FreePBX...`

The whole install process gets triggered by this one rather large if statement in `/etc/cont-init.d/10-freepbx`
````bash
if [ ! -f $WEBROOT/admin/index.php ]; then
	echo '** [freepbx] New Install Detected - Please wait while we fetch FreePBX...'
````
Basically just looking for the existence of that one index.php in the admin side of things.. Since you didn't delete your DB folder, easiest would be to move it out of the way, let the new install take its place, turn off the container and then copy the db back overtop of it. An `Apply Settings` will rewrite the freepbx files for you with the pertinent information. Sorry you are having problems with this. As I've found there are quite a few use cases out there each with their own unique issues.

We are having our own too with getting the UCP WebRTC Phone working nicely behind a reverse proxy - I have had my Systems Admin working on it for the past 2 weeks, and we are close and will share the steps involved to make this work.

**Comment by junalmeida** [comment]
followed your suggestion, no error at logs but: 

```
SQLSTATE[HY000] [1698] Access denied for user 'freepbxuser'@'localhost'::SQLSTATE[HY000] [1698] Access denied for user 'freepbxuser'@'localhost'
```

at the ui

**Comment by tiredofit** [comment]
The fun starts here :)
Try to stop the mysql service `service mysql stop`
Get the password statement from `/etc/obdc.ini`
Then make a file in `/tmp/reset_freepbx`
`SET PASSWORD FOR 'freepbxuser'@'localhost' = PASSWORD('thepasswordfromodbc.ini');`

Then, execute `mysqld --init-file=/tmp/reset_freepbx`

It'll go through a series of routines to reset the password for you, also probably a good idea to do the same thing for the root password (which is something random, and on the embedded version I can't recall at this time where they store it). Press Ctrl-C or exit the container and restart everything for it to be all back to normal and you should be back in FreePBX. Bit of a pain. When I first tried FreePBX in 2006 it seems they are using very similar install routines and desperately need to update it to todays standards..

Hope this helps!

**Comment by junalmeida** [comment]
I guess that I will need to recreate the database :(

**Comment by junalmeida** [comment]
my odbc.ini have no password

**Comment by tiredofit** [comment]
I strongly recommend the external DB hack I built - for precisely these moments, backups.. I added a new example in the docker-compose.yml recently that does automatic backups for you on a regular basis. This was a bit of a hairy image to build in the first place and I always feel a bit squeamish with having the DB inside all of the containers. This certainly is an image that takes a bit of getting used to.. 
Oddly enough, all of the PBX Images I have built (freepbx, fusionpbx, elastix, 3cx) are just as complicated.
